doc/developer/platform/reconciliation.md

```diff
@@ -0,0 +1,99 @@
+# Reconciling COMPUTE
+
+The controller and compute instances are logically related but can be hosted on different processes, and the state shared between the two needs to be reconciled.
+On restart, the controller wants to dictate the instances what their state should be.
```

Comment:
Briefly add some examples what the `state` contains

Reply:
I think it would help the reader for the next sections to give a bit a longer explanation of the components and how they manage state: Does compute come up clean-slate or does it have some locally persisted state?

And what are the scenarios do we consider, only compute restarts? Or also (temporary) network failure?